billing/models.py

```diff
@@ -1,3 +1,103 @@
+        EUR = CurrencyConstants.CurrencyDict["EUR"]["Code"]
+
+    class AccountingMethod(ChoiceEnum):
+        HOURLY = AccountingMethodText.HOURLY.value
+        MONTHLY = AccountingMethodText.MONTHLY.value
+
+    user = models.ForeignKey(CustomUser, on_delete=models.PROTECT, related_name="%(class)s")
+    value = models.PositiveSmallIntegerField(BillingDetailsAbstractModelText.VALUE.value, blank=False, null=False)
+    currency = models.CharField(
+        max_length=BillingConstants.CURRENCY_MAX_LENGTH.value, choices=Currency.choices(), default=Currency.PLN
+    )
+    accounting_method = models.CharField(
+        max_length=BillingConstants.ACCOUNTING_METHOD_MAX_LENGTH.value,
+        choices=AccountingMethod.choices(),
+        default=AccountingMethod.HOURLY.name,
+    )
+    from_date = models.DateTimeField(BillingDetailsAbstractModelText.FROM_DATE.value, blank=False, null=False)
+    to_date = models.DateTimeField(BillingDetailsAbstractModelText.TO_DATE.value, blank=True, null=True)
+
+    objects = BillingDetailsAbstractQuerySet.as_manager()
+
+    def clean(self) -> None:
+        super().clean()
+
+        if self.to_date is not None and self.to_date <= self.from_date:
+            raise ValidationError(
+                message=BillingValidationErrorText.VALIDATION_ERROR_FROM_DATE_LATER_THAN_TO_DATE.value
+            )
+        if self.to_date is None and not self.user.is_active:
+            raise ValidationError(
+                message=BillingValidationErrorText.VALIDATION_ERROR_TO_DATE_EMPTY_FOR_INACTIVE_USER.value
+            )
+
+    @property
+    def get_currency_name(self) -> str:
+        return CurrencyConstants.CurrencyDict[self.currency]["Name"]
+
+    @property
+    def get_currency_code(self) -> str:
+        return CurrencyConstants.CurrencyDict[self.currency]["Code"]
+
+    @property
+    def get_currency_symbol(self) -> str:
+        return CurrencyConstants.CurrencyDict[self.currency]["Symbol"]
+
+
+class UserSalaryDetailsManager(models.Manager):
+    def get_ongoing_salary_or_none(self, user: CustomUser) -> Union[models.QuerySet, None]:
+        try:
+            return self.get(user_id=user.id, to_date__isnull=True)
+        except self.model.DoesNotExist:
+            return None
+
+
+class UserSalaryDetails(BillingDetailsAbstract):
+    objects = UserSalaryDetailsManager()
+
+    def clean(self) -> None:
+        self._check_ongoing_salaries()
+        super().clean()
+
+    def _check_ongoing_salaries(self) -> None:
+        if self.to_date is None:
+            salary_details = UserSalaryDetails.objects.get_ongoing_salary_or_none(self.user)
+            if salary_details is not None:
+                current_time = timezone.datetime.now(tz=timezone.get_default_timezone())
+                salary_details.to_date = timezone.make_aware(
+                    datetime.datetime(year=current_time.year, month=current_time.month, day=current_time.day)
+                )
+                salary_details.save()
```

Comment:
What is the purpose of `_check_ongoing_salaries`. Why is it in clean? Why we check again only if `self.to_date is None` ?Why do we apply with this function `to_date` value?